src/components/UserInput/UserInput.js

```diff
@@ -0,0 +1,68 @@
+import { useState } from 'react';
+import './UserInput.scss';
+
+const UserInput = ({ name, koreanName, condition, loginInputHandler }) => {
+  const [loginValidationBtn, setLoginValidationBtn] = useState(false);
+
+  const validationChecker = condition => {
+    if (condition) {
+      setLoginValidationBtn(true);
+    } else {
+      setLoginValidationBtn(false);
+    }
+  };
+
+  const [inputTitleRedBtn, setInputTitleRedBtn] = useState(false);
+  const [inputTitleGoUpBtn, setInputTitleGoUpBtn] = useState(false);
+
+  const inputFocusEffector = e => {
+    setInputTitleGoUpBtn(true);
+  };
+
+  const inputBlurEffector = e => {
+    e.target.value ? setInputTitleGoUpBtn(true) : setInputTitleGoUpBtn(false);
+  };
+
+  const inputRedEffector = () => {
+    loginValidationBtn ? setInputTitleRedBtn(false) : setInputTitleRedBtn(true);
+  };
+  return (
+    <>
+      <div className="user-input">
+        <span
+          className={
+            'user-placeholder ' +
+            (inputTitleGoUpBtn ? 'input-title-goup' : null)
+          }
+          style={inputTitleRedBtn ? { color: 'red' } : { color: 'gray' }}
+        >
+          {koreanName} *
+        </span>
+        <input
+          type={name}
+          name={name}
+          className={inputTitleRedBtn ? 'input-border-red' : null}
+          onFocus={e => {
+            inputFocusEffector(e);
+            inputRedEffector();
+          }}
+          onBlur={e => {
+            inputBlurEffector(e);
+            inputRedEffector();
+          }}
+          onChange={e => {
+            loginInputHandler(e);
+            validationChecker({ condition });
+          }}
```

Comment:
```suggestion
           onFocus={handleInputTitle}
          onBlur={handleInputTitle}
          onChange={e => {
            loginInputHandler(e);
          }}
```
condition으로 불필요한 state를 관리하고
focus와 blur를 하나의 함수로 처리한다면
이와 같은 형태로 구현이 되겠습니다.

Reply:
멘토님 질문이 있습니다 ! 기존에 인풋창의 onFocus 이벤트에 대해서 두가지 함수가 적용되고 있었는데요 ! (inputFocusEffector / inputRedEffector) 앞의 inputFocusEffector의 경우 **인풋창 안에 값이 있는지, 없는지 조건**에 따라 글자가 올라가고 내려가는 효과이고, 뒤의 RedEffector는 **유효성검사 결과**에 따라 글자가 빨갛게 보일지 검은색으로 보일 지를 결정하는 함수입니다 ! 각각 다른 조건에 따라 움직이기 때문에 하나로 묶을 때 경우의 수가 많아지게 되어서요, 다른 조건을 가지는 함수여도 하나의 함수로 묶어서 관리하는게 좋을지 궁금합니다 !